### PR TITLE
fix: check iptable rules match expected

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

Checking if the length of the iptable rules in the `aad-metadata` chain equals 3 is not sufficient as it's possible the rules were changed underneath. To ensure the rules are what we expect it to be, check if existing rules match all expected rules. If any difference in the rules, then flush and recreate the chain.

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

**Comment by chewong** [comment]
You could ignore the CI errors from the new framework. I included improvements in #656 & #662 which should reduce the error rate.

**Comment by aramase** [comment]
sounds good! Are the 2 PRs ready for review?

**Comment by chewong** [comment]
Both are ready for review but #656 should get merged first since #662 depends on it

File: pkg/nmi/iptables/iptables.go [hunk 1]
```diff
@@ -89,9 +89,28 @@ func ensureCustomChain(ipt *iptables.IPTables, destIP, destPort, targetip, targe
 
 		For this reason we check if the length of rules is 3. If not 3, then we flush and create chain again.
 	*/
-	if len(rules) == 3 {
+
```
Comment (chewong) [inline pkg/nmi/iptables/iptables.go:92]: nit: `s/10.240.3.134:2579/127.0.0.1:<nmi port>` in the comment block above.

File: pkg/nmi/iptables/iptables.go [hunk 2]
```diff
@@ -89,9 +89,28 @@ func ensureCustomChain(ipt *iptables.IPTables, destIP, destPort, targetip, targe
...
+	expectedRules := map[string]struct{}{
+		"-N aad-metadata": {},
+		"-A aad-metadata ! -s 127.0.0.1/32 -d 169.254.169.254/32 -p tcp -m tcp --dport 80 -j DNAT --to-destination 127.0.0.1:2579": {},
```
Comment (chewong) [inline pkg/nmi/iptables/iptables.go:95]: `2579` is the default NMI port but it can be changed via `--nmi-port` flag. Let's account that scenario as well.

**Pull request merged**